src/util/TimeUtil.js

```diff
@@ -93,4 +93,31 @@ export default {
       unit: convertMomentUnits(highestUnit),
     };
   },
+
+  /**
+   * @method formatDateToDeviceAssuranceGracePeriodExpiryLocaleString
+   * Conversion from a Date object to a locale string that mimics Okta's `short-with-timezone` format
+   * e.g. new Date(2024-09-05T00:00:00.000Z) -> 09/05/2024, 8:00 PM EDT
+   *
+   * @param {Date} date The Date object for the grace period expiry
+   * @param {LanguageCode} languageCode The user's language code / locale
+   * @return {string} The formatted `short-with-timezone` local string
+   */
+  formatDateToDeviceAssuranceGracePeriodExpiryLocaleString: (date, languageCode) => {
+    try {
+    // Invalid Date objects will return NaN for valueOf()
+      return date && !isNaN(date.valueOf()) && languageCode !== null ? date.toLocaleString(languageCode, {
+        year: 'numeric',
+        month: '2-digit',
+        day: '2-digit',
+        hour: 'numeric',
+        minute: 'numeric',
```

Comment:
I thought you were truncating the `minute` -- or maybe you meant you'd round down? I thought `minute: 'numeric'` would show a single digit for values <10m after the hour, but it seems to be overridden by one of the other values

Reply:
Ah, yeah we're rounding down, though I actually forgot to enforce that. I looked up why `numeric` doesn't show a single digit value and apparently that happens for specific combinations of these options, but to be safe I'll use '2-digit' for `hour` and `minute`